### ISSUE TITLE
Training instability?

### ISSUE THREAD

**Issue by kdmayer** [lucidrains/equiformer-pytorch#24]
Hi, 

Thanks for the great work, @lucidrains!

I've been playing around with your Equiformer implementation as an encoder for a 3D graph. When looping over all my data samples, the encoding works fine, but as soon as I add weight updates with optimizer.step(), it only takes a couple batches until the encoder starts to produce NaN values for out.type0. 

I've tried different optimizers, i.e., Adam and SGD, but the problem persists.

Do you have an idea where this instability might come from and what I could do?

Thank you,
Kevin

**Comment by lucidrains** [comment]
thanks Kevin!

how high is your learning rate? also, try adding gradient clipping, which is pretty standard for transformer training

**Comment by lucidrains** [comment]
@kdmayer also, just so you know, there is a new sota for this line of research https://github.com/lucidrains/gotennet-pytorch , without any tensor products and clebsh gordan

**Comment by kdmayer** [comment]
It looks like the observed training instability came from the way I defined my variational graph autoencoder architecture initially. After playing around a bit more, I am now able to train my baseline model :)

Will close the issue. Thanks for the pointer!

**Comment by lucidrains** [comment]
@kdmayer no problem, pay attention to gotennet

got a couple material science researchers already interested
